test/helpers/kubectl.go

```diff
@@ -878,6 +896,24 @@ func (kub *Kubectl) ciliumInstall(dsPatchName, cmPatchName string, getK8sDescrip
 	if err := kub.DeployPatch(dsPathname, getK8sDescriptorPatch(dsPatchName)); err != nil {
 		return err
 	}
+
+	if versionRequiresOperator(versionTag) {
+		ginkgoext.By("Deploying Cilium operator")
+		if err := deployOriginal(operatorRbacPathname); err != nil {
+			return err
+		}
+
+		if err := kub.DeployPatch(operatorPathname, getK8sDescriptorPatch(operatorPatchName)); err != nil {
+			return err
+		}
+
+		ginkgoext.By("Waiting for operator to become ready")
+		err := kub.WaitforNPods(KubeSystemNamespace, "-l name=cilium-operator", 1, 300)
```

Comment:
I would not wait here, for wait I would add that into `ExpectCiliumReady`  due is master all needs the operator. 

Also, I would create `ExpectCiliumReadyVersion(version)` to validate in the upgrade test here: 
https://github.com/cilium/cilium/blob/master/test/k8sT/Updates.go#L152

The reason, in some cases we install Cilium + ETCD operator + dns and we install all, and wait after that. In this case can produce a chicken egg issue. 

Just a comment, feel free to change or not.

Reply:
What is the reason for the split? I understand that everything has to be deployed before we start waiting for things to come up but why does the readiness check have to be called separately. What if the user forgets?

Reply:
My main point is that it is quite hard to understand that `ExpectCiliumReady()` has to be called afterwards but only after `DeployETCDOperator()` has been called first. I don't fully understand why these functions are split apart and why we don't just have a single install function that ensures proper installation and fails otherwise.

Anyway, I've removed the check as it is mostly cosmetic. The operator can come up in the back while the tests are already running.